addons/stock/static/src/widgets/generate_serial.js

```diff
@@ -51,6 +51,9 @@ export class GenerateDialog extends Component {
                 })
             );
         }
+        lines.records.slice(0, newlines.length).map((record) => {
+            lines.delete(record);
+        });
```

Comment:
I  think having 
` lines.records = newlines;` instead of `push` is enough

Reply:
doesn't work:
If I remove `lines.records.push` but keep `lines._commands.push`, I will have 5 lines on the popup but 10 lines (5 with serial, 5 without) in detailed operations after saving.
If I remove both `push`, no change is applied in detailed op after saving.
I'm looking into createAndReplace to see if it solves the issue, but baring that, the delete looks to be the best solution for now.

Reply:
yeah no, `createAndReplace` is not adapted for that. But I replaced `records.push` by a `Promise.all` to make the transition seamless